src/it/resources-noncompilable/org/checkstyle/suppressionxpathfilter/constructorsdeclarationgrouping/InputXpathConstructorsDeclarationGroupingRecords.java

```diff
@@ -0,0 +1,20 @@
+//non-compiled with javac: Compilable with Java14
+
+package org.checkstyle.suppressionxpathfilter.constructorsdeclarationgrouping;
+
+public class InputXpathConstructorsDeclarationGroupingRecords {
+  public record MyRecord(int x, int y) {
+
+    public MyRecord(int a) {
+      this(a,a);
+    }
+
+    void foo() {}
+
+    public MyRecord {} // warn
+
+    public MyRecord(int x, int y, int z) {
+      this(x+y, z);
+    }
```

Comment:
@Zopsss @romani it looks like we are emulating the behavior of OverloadMethodsDeclarationOrder where a user may need to run the check multiple times in order to make checkstyle happy. Is there some technical limitation that prevents us from using the following logic:

 - First constructor is where constructors should live
 - All other constructors are flagged

?

Reply:
yes we can do this, I agree with this. Waiting for @romani's opinion. If he agrees then I'll start working on the new implementation

Reply:
@nrmancuso @romani If this is the way to want to move, we should probably create an issue for the other check.

Reply:
@nrmancuso I've changed the implementation according to your new idea. Should I push the new changes or wait for other maintainers approval?

Reply:
> @nrmancuso I've changed the implementation according to your new idea. Should I push the new changes or wait for other maintainers approval?

If you’ve already done the work, you can checkout a new branch from what you’ve done, and open a draft PR. Then you can reset this branch from the origin. This way you have both versions preserved and we can see what the other looks like.

Reply:
Done. Here is the draft PR: https://github.com/checkstyle/checkstyle/pull/14844

Reply:
>Done. Here is the draft PR: https://github.com/checkstyle/checkstyle/pull/14844

This new PR makes it even more obvious that we should do [this](https://github.com/checkstyle/checkstyle/pull/14749#discussion_r1583033202), IMO.

This is my vision:
1. Create a list of all children of the class/enum/whatever def
2. Create sublist from first constructor , taking while the AST is a constructor
3. Take the difference of the original list and the sublist
4. Filter out non-constructors
5. log all remaining items

Reply:
made the implementation stateless on that PR, can you please check that out? and sorry for the late reply I wasn't active for few days